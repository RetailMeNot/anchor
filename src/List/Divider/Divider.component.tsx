// REACT
import * as React from 'react';
// VENDOR
import classNames from 'classnames';
import styled from '@xstyled/styled-components';
import { space as spaceStyles, SpaceProps } from '@xstyled/system';

export interface DividerProps extends React.HTMLAttributes<HTMLDivElement>, SpaceProps {
    onMouseOver?: (event: React.MouseEvent) => any;
    onMouseOut?: (event: React.MouseEvent) => any;
    onSelect?: (...props: any) => any;
    active?: boolean;
    value?: string | number;
    children?: any;
    className?: string;
}

const StyledDivider = styled('div')<DividerProps>`
    height: 0.5rem;
    padding: 0.5rem 1rem;
    hr {
        border: none;
        border-top: light;
    }
    ${spaceStyles}
`;

export const Divider = ({
    className,
    ...props
}: DividerProps): React.ReactElement<any> => (
    <StyledDivider
        className={classNames('anchor-list-divider', className)}
        {...props}
    >
        <hr />
    </StyledDivider>
);
