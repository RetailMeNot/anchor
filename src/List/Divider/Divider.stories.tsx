// REACT
import * as React from 'react';
// STORYBOOK
import { storiesOf } from '@storybook/react';
// VENDOR
import styled from 'styled-components';
import { ThemeProvider } from '@xstyled/styled-components';
// COMPONENTS
import { Divider } from './Divider.component';
import { RootTheme } from '../../theme';

const StyledStory = styled('div')`
    padding: 2rem 5rem;
`;

storiesOf('Components/List/Divider', module).add('Primary', () => (
    <ThemeProvider theme={RootTheme}>
        <StyledStory>
            <p>Divider</p>
            <Divider>Divider 1</Divider>
        </StyledStory>
    </ThemeProvider>
));
