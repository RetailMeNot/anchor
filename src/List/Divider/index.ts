export { Divider } from './Divider.component';
