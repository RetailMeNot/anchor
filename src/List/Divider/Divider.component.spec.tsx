// REACT
import * as React from 'react';
import * as renderer from 'react-test-renderer';
// VENDOR
import { shallow, mount } from 'enzyme';
// COMPONENT
import { Divider } from './Divider.component';

describe('Component: Divider', () => {
    it('should be defined', () => {
        const subject = <Divider />;
        const wrapper = mount(subject);
        const component = shallow(subject);

        expect(subject).toBeDefined();
        expect(wrapper).toBeDefined();
        expect(component).toBeDefined();
        const tree = renderer.create(subject).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
