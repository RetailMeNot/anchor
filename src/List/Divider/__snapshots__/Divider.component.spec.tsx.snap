// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Divider should be defined 1`] = `
.c0 {
  height: .5rem;
  border-top: solid thin #D3D3D3;
  margin-top: 1rem;
}

<div
<<<<<<< HEAD
  className="anchor-list-divider c0"
=======
  className="anchor-list-divider sc-cpmLhU gjzPVC"
>>>>>>> chore(Button): update snapshots due to modifying theme file
/>
`;
